backend/stylists/app/Http/Controllers/RecommendationController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use App\Http\Requests;
+use Illuminate\Support\Facades\Auth;
+use Validator;
+use App\Recommendation;
+use App\Push;
+use App\Client;
+use App\Stylist;
+use App\Look;
+use App\Product;
+use App\Models\Lookups\EntityType;
+use App\Models\Enums\AppSections;
+use App\Models\Enums\RecommendationType;
+use App\Models\Enums\EntityType as EntityTypeId;
+
+class RecommendationController extends Controller
+{
+    public function index(Request $request, $action, $id = null, $action_id = null)
+    {
+        $method = strtolower($request->method()) . strtoupper(substr($action, 0, 1)) . substr($action, 1);
+        if ($id) {
+            $this->resource_id = $id;
+        }
+        return $this->$method($request);
+    }
+
+    public function postSend(Request $request)
+    {
+        $app_section = isset($request->app_section) && $request->app_section != "" ? $request->app_section : AppSections::MY_REQUESTS;
+
+        $entity_type_id = '';
+        if (isset($request->entity_type_id)) {
+            $entity_type_id = $request->entity_type_id;
+        } else {
+            return response()->json(
+                array(
+                    'error_message' => 'Entity type undefined'
+                ), 200
+            );
+        }
+
+        $client_ids = isset($request->client_ids) ? $request->client_ids : '';
+        $entity_ids = isset($request->entity_ids) ? $request->entity_ids : '';
+
+        $stylish_id = Auth::user()->stylish_id;
+        $entity_type_data = EntityType::where('id', $entity_type_id)->first();
+        if(empty($entity_type_data)){
+            return response()->json(
+                array(
+                    'error_message' => 'Entity type name not found'
+                ), 200
+            );
+        }
+
+        $stylish_data = Stylist::where('stylish_id', $stylish_id)->first();
+        if(empty($stylish_data)){
+            return response()->json(
+                array(
+                    'error_message' => 'Invalid stylist'
+                ), 200
+            );
+        }
+
+        $client_data = Client::whereIn('user_id', $client_ids)->get();
+        if (empty($client_data)) {
+            return response()->json(
+                array(
+                    'error_message' => 'Client not found'
+                ), 200
+            );
+        }
+
+        $entity_data = '';
+        if ($entity_type_id == EntityTypeId::PRODUCT) {
+            $entity_data = Product::whereIn('id', $entity_ids)->get();
+        } elseif ($entity_type_id == EntityTypeId::LOOK) {
+            $entity_data = Look::whereIn('id', $entity_ids)->get();
+        }
+        if (empty($entity_data)) {
+            return response()->json(
+                array(
+                    'error_message' => 'Entity not found'
+                ), 200
+            );
+        }
+
+        sort($entity_ids);
```

Comment:
Dont use entity_ids and client_ids after this step.

Use entity_data and client_data